lib/backend/model/block_affinity.go

```diff
@@ -52,13 +52,24 @@ func (key BlockAffinityKey) valueType() reflect.Type {
 }
 
 type BlockAffinityListOptions struct {
-	Host string
+	Host      string
+	IPVersion int
 }
 
 func (options BlockAffinityListOptions) DefaultPathRoot() string {
 	k := "/calico/ipam/v2/host/"
 	if options.Host != "" {
 		k = k + options.Host
+
+		if options.IPVersion != 0 {
+			k = k + fmt.Sprintf("/ipv%d/block", options.IPVersion)
+		}
+	} else {
+		// If no host is specified, the IP version must also
+		// be unspecified.
+		if options.IPVersion != 0 {
```

Comment:
Yeah I guess I could do that, but I didn't need that function so I didn't do it and just added this so that no-one was surprised when they tried to get all IPv4 block affinities and it didn't work.